typescript/hacker-news-agent/src/api.ts

```diff
@@ -0,0 +1,83 @@
+const HN_SEARCH_URL = "https://hn.algolia.com/api/v1/search";
+const API_TIMEOUT = 30000;
+
+export interface HackerNewsStory {
+  objectID: string;
+  title: string;
+  url?: string;
+  points: number;
+  num_comments: number;
+  author: string;
+  created_at: string;
+}
+
+export interface HackerNewsComment {
+  objectID: string;
+  comment_text: string;
+  author: string;
+  created_at: string;
+  story_id: string;
+}
+
+export async function searchHackerNews(
+  query: string,
+  maxResults: number = 20,
+): Promise<HackerNewsStory[]> {
+  const params = new URLSearchParams({
+    query,
+    hitsPerPage: maxResults.toString(),
+    tags: "story",
+  });
+
+  try {
+    const controller = new AbortController();
```

Comment:
Looks like `AbortController` is the older answer. The new way to do this is abort signal: `fetch(url, { signal: AbortSignal.timeout(5000) })`
https://stackoverflow.com/a/50101022